Sources/EmbraceCore/Capture/Network/URLSessionTaskHandler.swift

```diff
@@ -121,14 +127,30 @@ final class DefaultURLSessionTaskHandler: URLSessionTaskHandler {
 
     func finish(task: URLSessionTask, data: Data?, error: (any Error)?) {
         queue.async {
+            // save end time for payload capture
+            task.embraceEndTime = Date()
+
+            // process payload capture
+            self.payloadCaptureHandler.process(
```

Comment:
I wonder if we should pass in the OTel SpanContext to this captured payload to link directly to the span we recorded?

If we do that the startTime/endTime seems redundant